### ISSUE TITLE
Add Cancel support to options update.

### ISSUE THREAD

**Issue by PhotoNomad0** [unfoldingWord-dev/ts-android#1689]
Add Cancel support to options update. 

see #1677 
and https://github.com/unfoldingWord-dev/ts-requirements/issues/189

**Comment by cckozie** [comment]
Canceling does work, but the user messaging is inconsistent.
Canceling the Update List of Available Source Texts and Update List of Available Target Languages both display a message indicating that the update was canceled. There is no such error message displayed when the Down Available Source Texts is canceled.

**Comment by cckozie** [comment]
There is now a message when the downloading of source texts is canceled, but the information provided is not always complete/accurate. 
I deleted all 4 entries for Matthew from the resource_containers folder. I then selected English UDB and ULB for Matthew to be downloaded. Shortly after starting the download I clicked the cancel button. After several seconds the app reported that 1 source text had been successfully downloaded and the Matthew udb displayed a green check and the ulb still displays the checked box. That is all good.
I then clicked the download button again and after a few seconds clicked the cancel button. After a few seconds the app again reported 1 source text successfully downloaded, and now shows the green check by the ulb. This is somewhat misleading as the tN and tQ sources had not yet been downloaded.
I repeated the process and this time got a message that says that the tN download failed, and neither tN or tQ had been downloaded at that point.
When a ulb is selected for download and the download is canceled before all associated files have successfully been downloaded, the message should indicate that not all sources were successfully downloaded and the green check should not be applied to the source.
Also, since 4 files are downloaded when the user selects a ulb, why does the message indicate 3 sources downloaded?
